systemtest/src/main/java/io/strimzi/systemtest/utils/kafkaUtils/KafkaUtils.java

```diff
@@ -27,19 +28,23 @@ private KafkaUtils() {}
 
     public static void waitUntilKafkaCRIsReady(String clusterName) {
         LOGGER.info("Waiting till Kafka CR will be ready");
+        Condition condition = Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0);
+
         TestUtils.waitFor("Waiting for Kafka resource status is ready", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
-            () ->   Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0).getType().equals("Ready") &&
-                Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0).getStatus().equals("True")
+            () ->   condition.getType().equals("Ready") && condition.getStatus().equals("True")
         );
         LOGGER.info("Kafka CR will be ready");
     }
 
-    public static void waitUntilKafkaStatusConditionIsPresent(String clusterName) {
-        LOGGER.info("Waiting till kafka resource status is present");
-        TestUtils.waitFor("Waiting for Kafka resource status is ready", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
-            () ->  Crds.kafkaOperation(kubeClient().getClient()).inNamespace(kubeClient().getNamespace()).withName(clusterName).get().getStatus().getConditions().get(0) != null
+    public static void waitUntilKafkaStatusConditionIsNotReady(String clusterName) throws InterruptedException {
```

Comment:
Sorry to be nitpicking ... but some more inconsistencies ...
* Few lines above you have method `waitUntilKafkaCRIsReady` which does the same just for `Ready` state. Could we have some naming consistency?
* The method above doesn't seem to need the `Thread.sleep(...)`. Why do we need it here?

Reply:
- First point solved.  
- Second point, previously i did some experiments. Actually, i need there `Thread.sleep(....)`, because the default status when first appears correctly i was getting `Null Pointers exception` for some reason.

Reply:
Basically, this one without explicitly waiting. 
```
Expected :a string containing "Version 6.6.6 is not supported."
Actual   :"Kafka cluster is being deployed"
```

Reply:
Wait ... that is silly solution. They way this works is following:
* When you deploy the new resource we first thing off set a new status that we are deploying with NotReady=true.
* We start the reconciliation
* In this case, we fail pretty quickly because of configuration error and set the status to NotReady=true with new message.

Now I can see why the 1000ms sleep helps you. But
* It might not be always 1000ms.
* Not in every case this is needed. If you would have usecase which tests some NotReady=true state on running cluster, this 1000ms sleeep would be useless because the initial state would be not invovled.

So I think you should do one of these:
* Move the sleep into the test cases from the wait method (still kind stupid)
* Change the method to something liek `waitForNotReadyAndMessage` and not wait just for the NotReady state but also for some specific message to go with it.

Reply:
I modify method with also waiting on concrete message and it seems that everything is correct. Thanks :)